### ISSUE TITLE
Identity UI Password validation has string length is hardcoded to 6

### ISSUE THREAD

**Issue by sherif-elmetainy** [dotnet/aspnetcore#25788]
### Describe the bug
In the Microsoft.AspNetCore.Identity.UI package, the StringLength validation for the password in registration form is [hardcoded to 6](https://github.com/dotnet/aspnetcore/blob/v3.1.8/src/Identity/UI/src/Areas/Identity/Pages/V4/Account/Register.cshtml.cs).
So changing the `PasswordOptions.RequiredLength` property in Startup.cs has no effect on the validation.

And you can look in [Source Code](https://github.com/dotnet/aspnetcore/blob/v3.1.8/src/Identity/UI/src/Areas/Identity/Pages/V4/Account/Register.cshtml.cs) that `Password` property has the following attributes:

```csharp
[Required]
[StringLength(100, ErrorMessage = "The {0} must be at least {2} and at max {1} characters long.", MinimumLength = 6)]
[DataType(DataType.Password)]
[Display(Name = "Password")]
public string Password { get; set; }
```
Which is as you can see the length is hardcoded.

The [Reset Password](https://github.com/dotnet/aspnetcore/blob/v3.1.8/src/Identity/UI/src/Areas/Identity/Pages/V4/Account/ResetPassword.cshtml.cs) and [Change Password](https://github.com/dotnet/aspnetcore/blob/v3.1.8/src/Identity/UI/src/Areas/Identity/Pages/V4/Account/Manage/ChangePassword.cshtml.cs) also have the same hardcoded value.

### To Reproduce
1. Create a new Asp.NET Core Web Application with following options: WebApplication (Model-View-Controller) and Individual User Account calculator.
2. In startup.cs change identity options as follows:
```csharp
services.AddDefaultIdentity<IdentityUser>(options =>
{
    options.SignIn.RequireConfirmedAccount = true;
    options.Password.RequiredLength = 4; // This line is changing the default
}).AddEntityFrameworkStores<ApplicationDbContext>();
```
3. Run the application, go the the Register Page, Enter a 4 letter password.

### Expected Output
Password of length 4 chars or more should be accepted to match the setting for `RequiredLength` property.
### Actual Output
Password validation fails and the validation message indicates minimum length is 6 characters.

### Exceptions (if any)
No exceptions are thrown

### Further technical details
- ASP.NET Core version: 3.1.8
- dotnet --info 
```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.402
 Commit:    9b5de826fd

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.402\

Host (useful for support):
  Version: 3.1.8
  Commit:  9c1330dedd

.NET Core SDKs installed:
  3.1.401 [C:\Program Files\dotnet\sdk]
  3.1.402 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

- Visual Studio 2019 Version 16.7.3

**Comment by blowdart** [comment]
@HaoK validation attributes need to be constants at compile time do they not? So we can't affect the form ui without a custom validation attribute?

**Comment by HaoK** [comment]
Yup, I think that's the main issue that it needs to be a constant

**Comment by blowdart** [comment]
@sherif-elmetainy Do you really want such a short password? What's the use case here? I'm inclined to just mark this as won't fix, but I'd like to know your needs here

**Comment by sherif-elmetainy** [comment]
> @sherif-elmetainy Do you really want such a short password? What's the use case here? I'm inclined to just mark this as won't fix, but I'd like to know your needs here

@blowdart . The 4 in the issue was just an example. The issue applies whether `RequiredLength ` is set to 4 or 8 or whatever.  The validator behavior doesn't match the `RequiredLength` setting for any value except for the hardcoded 6.

Let's say I set to minimum Length to 8. User enters a 5 char password. Validator client side message will tell him/her length must be 6 or more. He/she enters a 6 letter password, client validation will pass, submit form, then the server update will fail and now he gets a message that it's actually 8. It's confusing when users gets a message that min length is 6, then enters a 6 letter password then the server replies "sorry I meant 8".

So it is not about me wanting a short password or a long password. It's about weird behavior.

**Comment by blowdart** [comment]
Hmm probably best to remove it entirely then. We'll look at this for 6.0

**Comment by HaoK** [comment]
Yeah in the old days when the attribute was in the app, it was at least possible for them to keep in sync, getting rid of the client side validation seems better I guess

**Comment by sherif-elmetainy** [comment]
> Hmm probably best to remove it entirely then. We'll look at this for 6.0

Yes, I think removing the attribute entirely might be a good idea. I just looked at source code, and found that if the developer injects a custom service implementing `IPasswordValidator`, the entire `PasswordOptions` object will not be checked at all. So a custom attribute will have to be server side only which is already implemented without an attribute.
